src/ui/users/add/AddFormContainer.js

```diff
@@ -12,26 +12,42 @@ import { ROUTE_USER_LIST } from 'app-init/router';
 
 import UserForm from 'ui/users/common/UserForm';
 
-export const mapStateToProps = state => ({
-  profileTypes: getProfileTypesOptions(state),
-});
-
-export const mapDispatchToProps = (dispatch, { history }) => ({
-  onSubmit: (user) => {
-    const { saveType } = user;
-    dispatch(sendPostUser(user, saveType === 'editProfile'));
-  },
-  onWillMount: () => {
-    dispatch(destroy('user'));
+const AddFormContainer = () => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const profileTypes = useSelector(getProfileTypesOptions);
+
+  const handleMount = useCallback(() => {
```

Comment:
any reason why you have changed the format? we usually code this kind with `mapStateToProps` and `mapDispatchToProps`

Reply:
We're gradually introducing hooks into our codebase. And writing our components using hooks makes it more future-proof since, of course, that's react's future, which has been followed by all prominent libraries including react-redux. [Using react-redux hooks also has a few other benefits](https://stackoverflow.com/questions/59748180/should-i-use-useselector-usedispatch-instead-of-mapstatetoprops#answer-66218178). And I personally prefer that for a few other reasons (i.e. component-based, enforces more reliable testing, etc.).